### ISSUE TITLE
splitHaplotype core dump

### ISSUE THREAD

**Issue by gconcepcion** [marbl/canu#1242]
```
${CANU_HOME}/canu -p hg002_trio \
     -d hg002_trio \
     -haplotype \
     genomeSize=3200m \
     gridEngineResourceOption="-pe smp THREADS -l mem_free=MEMORY" \
     gridOptions="-V -q bigmem " \
     -haplotypeHG003 /pbi/dept/appslab/datasets/wr_HG002_NIST_downloads/trio/HG003/D2_S*.fastq.gz \
     -haplotypeHG004 /pbi/dept/appslab/datasets/wr_HG002_NIST_downloads/trio/HG004/D3_S*.fastq.gz \
     -pacbio-corrected ${INPUT}
```
```
$ ~/src/github/canu/Linux-amd64/bin/canu --version
Canu snapshot v1.8 +117 changes (r9327 dc859c7b4d2065d9412d5683e71d289af6ebf7ed)
```

```
LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.2.1511 (Core) 
Release:        7.2.1511
Codename:       Core
```

It's my first time running trio binning on this PacBio HG002 CCS dataset, but i'm getting a core dump in splitHaplotypes.sh. It's entirely possible i'm doing something wrong, but the ecoli example trio binning worked without any problems. This is similar (though not exactly the same) data that was used in the recent BioRXIV pre-print on Highly accurate long reads (CCS) that came out last month 

```
$ cat 1.out 

Found perl:
   /usr/bin/perl
   This is perl 5, version 16, subversion 3 (v5.16.3) built for x86_64-linux-thread-multi

Found java:
   /mnt/software/j/jdk/1.8.0_60/bin/java
   java version "1.8.0_60"

Found canu:
   /home/UNIXHOME/gconcepcion/src/github/canu/Linux-amd64/bin/canu
   Canu snapshot v1.8 +117 changes (r9327 dc859c7b4d2065d9412d5683e71d289af6ebf7ed)

Running job 1 based on SGE_TASK_ID=1 and offset=0.
splitHaplotype: utility/kmers-exact.C:399: void kmerCountExactLookup::load(kmerCountFileReader*): Assertion `_suffixBgn[ii] == _suffixEnd[ii-1]' failed.

Failed with 'Aborted'; backtrace (libbacktrace):
utility/system-stackTrace.C::89 in _Z17AS_UTL_catchCrashiP7siginfoPv()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
utility/kmers-exact.C::399 in _ZN20kmerCountExactLookup4loadEP19kmerCountFileReader()
utility/kmers.H::703 in _ZN20kmerCountExactLookupC4EP19kmerCountFileReaderjj()
haplotyping/splitHaplotype.C::404 in _ZN7hapData19initializeKmerTableEv()
haplotyping/splitHaplotype.C::459 in _ZN7allData17loadHaplotypeDataEv()
haplotyping/splitHaplotype.C::771 in main()
(null)::0 in (null)()
(null)::0 in (null)()
/var/spool/gridengine/execd/mp1201-sge66/job_scripts/714563: line 88:  8632 Aborted                 (core dumped) /home/UNIXHOME/gconcepcion/src/github/canu/Linux-amd64/bin/splitHaplotype -threads 16 -R /pbi/dept/bifx/gconcepcion/190208eightM/HG002_8m_Q20.fastq -H ./0-kmers/haplotype-HG003.meryl ./0-kmers/reads-HG003.statistics ./haplotype-HG003.fasta.gz -H ./0-kmers/haplotype-HG004.meryl ./0-kmers/reads-HG004.statistics ./haplotype-HG004.fasta.gz -A ./haplotype-unknown.fasta.gz > haplotype.log.WORKING
```
```
$ ls -lt
total 161642128
-rw-r--r-- 1 gconcepcion Domain Users           20 Feb 12 21:06 haplotype-HG003.fasta.gz
-rw-r--r-- 1 gconcepcion Domain Users           20 Feb 12 21:06 haplotype-HG004.fasta.gz
-rw-r--r-- 1 gconcepcion Domain Users           20 Feb 12 21:06 haplotype-unknown.fasta.gz
-rw------- 1 gconcepcion Domain Users 164970639360 Feb 12 21:06 core.6316
-rw-r--r-- 1 gconcepcion Domain Users          122 Feb 12 20:58 haplotype.log.WORKING
-rwxr-xr-x 1 gconcepcion Domain Users         2153 Feb 12 20:51 splitHaplotype.sh
-rw-r--r-- 1 gconcepcion Domain Users         1708 Feb 12 19:24 1.out
-rw-r--r-- 1 gconcepcion Domain Users           69 Feb 12 19:09 splitHaplotype.jobSubmit-01.out
-rwxr-xr-x 1 gconcepcion Domain Users          223 Feb 12 19:09 splitHaplotype.jobSubmit-01.sh
drwxr-xr-x 8 gconcepcion Domain Users        12288 Feb 12 19:09 0-kmers
```

Any idea what's going on here? I have run the md5 check sum on all of the ftp downloaded short read data for kmer binning and it all checks out so I don't think bad input data is an issue. Also it seems to be an issue w/ regards to kmer counting so maybe missing kmers for some reason?

Thanks for your insight!

**Comment by brianwalenz** [comment]
Oh no!  `Canu snapshot v1.8 +117 changes` is literally right in the middle of some significant changes to the kmer data structures.  Was this run started with that version of canu?

The easiest and safest option would be to back up to the 1.8 release and restart completely -- 'rm -rf' the current run.  All canu has done so far is kmer processing, and all of that is potentially bogus.

It might be possible to update to the latest git version and see if it'll resume.  I'd advise against it though: 1) we haven't validated that the tip code works in quite a while; 2) haplotype splitting is kmer heavy, and one of the changes made would wreak havoc on the method used here if it's flubbed; 3) I'm *very* concerned about the size of the core dump, it's about 10 times larger than I'd expect, which is hinting that something bad did happen.

For humor (or if you're very brave and want to keep with the tip code), what's in 1.out and haplotype.log.WORKING?

**Comment by gconcepcion** [comment]
1.out is posted above, and haplotype.log.WORKING is:

```
--
-- Loading haplotype data.
--  Haplotype './0-kmers/haplotype-HG003.meryl':
--   use kmers with frequency at least 13.
```

No worries though, I restarted with the v1.8 release and am currently crossing my fingers.

Thanks!

**Comment by gconcepcion** [comment]
FYI: I tried numerous things, here are the two most significant:

1) I started from scratch with v1.8 release - but for some reason I couldn't figure out how to get it to do the haplotype binning. It accepted my `-haplotypeXXX` flag inputs, but the the `-haplotype` flag hadn't yet been implemented in the v1.8 release; and for some reason it seemed to be skipping the trio binning step for me and just performing a regular assembly.

2) Back to the drawing board I grabbed the latest code from lastnight; rebuilt canu and the problem has been solved and I have trio binned assemblies. I'll be evaluating quality over the next few days; but everything seems to be in working order.

Thanks!
